src/ome_zarr_models/v04/multiscales.py

```diff
@@ -1,17 +1,144 @@
-from typing import Any
+from __future__ import annotations
+from collections import Counter
+from typing import Annotated, Any, Sequence, get_args
 
-from pydantic import Field, field_validator
+from pydantic import AfterValidator, Field, model_validator
 
 from ome_zarr_models.base import Base
-from ome_zarr_models.utils import _unique_items_validator
-from ome_zarr_models.v04.axes import Axis
+from ome_zarr_models.utils import duplicates
+from ome_zarr_models.v04.axes import Axis, AxisType
 from ome_zarr_models.v04.coordinate_transformations import (
     PathScale,
     PathTranslation,
     VectorScale,
     VectorTranslation,
 )
 from ome_zarr_models.v04.omero import Omero
+from pydantic_zarr.v2 import ArraySpec, GroupSpec
+import zarr
+
+VALID_NDIM = (2, 3, 4, 5)
+NUM_TX_MAX = 2
+
+
+def ensure_transform_dimensionality(
```

Comment:
Thoughts on public vs. private API of these validation functions? My inclintation is to go private to be on the safe side of being able to modify/fix them easily in the future, and then the public API is just using the classes.

Reply:
private sounds good!